### DISCUSSION TITLE
Problem with Playing Sound in Scene Viewer

### DISCUSSION THREAD

**Discussion by NadimSKanaan** [google/model-viewer#2093] [Q&A]
Hello,

I am trying to add sound to the AR experience I am working on and I noticed something weird. The sound works fine when I load it and the model from my Glitch test. But, it crashes when I use the model and sound in my folder. Here's the code:

THIS WORKS:
      `src="https://cdn.glitch.com/c1626c7c-7146-4f31-92d9-7532fdabda7b%2FCamelWithAudio.glb?v=1614937451228?sound=https://cdn.glitch.com/c1626c7c-7146-4f31-92d9-7532fdabda7b%2FcamelSound.ogg?v=1614937654438"`

THIS **DOESN'T** WORK:
      `src="Camel.glb?sound=camelSound.ogg"` 

What do you think is going wrong?

**Comment by elalish** [comment]
@tpsiaki Any ideas?

**Reply by tpsiaki** [reply]
It looks like in the version that's not working you're trying to pass the audio file as an url parameter attached to the model url. I wouldn't expect this to work - the full Scene Viewer intent that gets built should have a separate model & sound parameter on the intent url.

Have you tried the following?
```
src="Camel.glb"
sound="camelSound.ogg"
```

**Reply by elalish** [reply]
we just repack any url parameters into the intent parameters. It's generally working, so I wonder if sound has a different syntax than the other options?

**Reply by tpsiaki** [reply]
I thought at some point it was changed to attach url parameters to the model so that if the model url had intent params - I think related to firebase hosting - that these would get passed on to Scene Viewer.

**Reply by tpsiaki** [reply]
Looking at the code - the problem is that only the base model url gets expanded w/ the location so the camelSound.ogg param will be copied directly to the intent. In Scene Viewer we don't do relative paths to the sound file (and tbh it wouldn't really make sense to have it use the model url as the relative path anyway), so that's why it's failing to load.

**Reply by elalish** [reply]
Thanks! Sounds like an easy fix; we need to do a point release anyway, so I'll try to get this in.

**Reply by elalish** [reply]
This is fixed now in v1.6
